### PR TITLE
Add pre_post_actions to TestActions

### PR DESCRIPTION
**Pull request by sebastianv1**
Adds pre_post_action attributes to `test_action` for custom actions run on the test scheme.

### CODE REVIEW

**Comment by maxwellE** [comment]
Will this run the post test action even when the test fails? I think there is some additional flag on the scheme to enable that.

**Comment by sebastianv1** [comment]
@maxwellE You're referring to `runPostActionsOnFailure` which I believe is still an undocumented feature. I don't see any flag in the UI.

File: xcodeproj/internal/xcode_schemes_internal.bzl [hunk 1]
```diff
@@ -136,6 +138,10 @@ def _test_action(
         expand_variables_based_on: Optional. One of the specified test target labels.
             If no value is provided, one of the test targets will be selected.
             If no expansion context is desired, use the `string` value `none`.
+        pre_actions: A `sequence` of `struct` values as created by
+            `xcode_schemes.pre_post_action`.
+        post_actions: A `sequence` of `struct` values as created by
+            `xcode_schemes.pre_post_action`.
```
Comment (brentleyjones) [inline xcodeproj/internal/xcode_schemes_internal.bzl:144]:
```suggestion
        pre_actions: Optional. A `sequence` of `struct` values as created by
            `xcode_schemes.pre_post_action`.
        post_actions: Optional. A `sequence` of `struct` values as created by
            `xcode_schemes.pre_post_action`.
```

File: xcodeproj/internal/xcode_schemes.bzl [hunk 2]
```diff
@@ -367,6 +413,10 @@ def make_xcode_schemes(bazel_labels):
                 target labels. If no value is provided, one of the test targets
                 will be selected. If no expansion context is desired, use the
                 `string` value `none`.
+            pre_actions: A `sequence` of `struct` values as created by
+                `xcode_schemes.pre_action`.
+            post_actions: A `sequence` of `struct` values as created by
+                `xcode_schemes.post_action`.
```
Comment (brentleyjones) [inline xcodeproj/internal/xcode_schemes.bzl:419]:
```suggestion
            pre_actions: Optional. A `sequence` of `struct` values as created by
                `xcode_schemes.pre_post_action`.
            post_actions: Optional. A `sequence` of `struct` values as created by
                `xcode_schemes.pre_post_action`.
```

**Comment by maxwellE** [comment]
> @maxwellE You're referring to `runPostActionsOnFailure` which I believe is still an undocumented feature. I don't see any flag in the UI. It will not run on failure AFAIK, and should match the existing pre/post actions for the build_action.

Sounds good, thanks

**Review by brentleyjones — APPROVED** [review]

**Comment by brentleyjones** [comment]
Thanks!

**Pull request merged**